modules/meshrenderinggl/include/modules/meshrenderinggl/processors/meshrasterizer.h

```diff
@@ -1,306 +1,315 @@
+        Off,  // no light, no reflection, just diffuse
+        Phong,
+        Pbr
+    };
+    enum class HatchingMode : char { U, V, UV };
+    enum class HatchingBlendingMode : char { Multiplicative, Additive };
+    /**
+     * \brief Hatching settings. These are exactly the parameters from the IRIS-paper
+     */
+    struct HatchingSettings {
+        HatchingSettings();
+        BoolCompositeProperty hatching_;
+
+        TemplateOptionProperty<HatchingMode> mode_;
+        IntProperty steepness_;
+        IntProperty baseFrequencyU_;
+        IntProperty baseFrequencyV_;
+
+        BoolCompositeProperty modulation_;
+        TemplateOptionProperty<HatchingMode> modulationMode_;
+        FloatProperty modulationAnisotropy_;
+        FloatProperty modulationOffset_;
+        FloatVec3Property color_;
+        FloatProperty strength_;
+        TemplateOptionProperty<HatchingBlendingMode> blendingMode_;
+    };
+    /**
+     * \brief The render settings per face.
+     * faceSettings_[0]=front face, faceSettings_[1]=back face
+     */
+    struct FaceSettings {
+        FaceSettings(bool frontFace);
+
+        bool frontFace_;
+        BoolCompositeProperty show_;
+        BoolProperty sameAsFrontFace_;
+        ButtonProperty copyFrontToBack_;
+
+        TransferFunctionProperty transferFunction_;
+        FloatVec3Property externalColor_;
+        TemplateOptionProperty<ColorSource> colorSource_;
+
+        BoolProperty separateUniformAlpha_;
+        FloatProperty uniformAlpha_;
+        TemplateOptionProperty<ShadingMode> shadingMode_;
+
+        BoolProperty showEdges_;
+        FloatVec3Property edgeColor_;
+        FloatProperty edgeOpacity_;
+
+        HatchingSettings hatching_;
+
+        // to copy front to back:
+        FaceSettings* frontPart_;
+
+        void copyFrontToBack();
+
+        void setUniforms(Shader& shader, std::string_view prefix) const;
+
+        bool lastOpaque_;
+    };
+
+    std::array<FaceSettings, 2> faceSettings_;
+
+    std::vector<std::shared_ptr<const Mesh>> enhancedMeshes_;
+
+    std::shared_ptr<Shader> shader_;
+
+    /**
+     * \brief This flag is set to true if adjacency information is available in the shader.
+     */
+    bool meshHasAdjacency_;
+    bool supportsFragmentLists_;
+};
+
+/**
+ * \brief Functor object that will render into a fragment list.
+ */
+class MeshRasterization : public Rasterization {
+public:
+    /**
+     * \brief Copy all settings and the shader to hand to a renderer.
+     */
+    MeshRasterization(const MeshRasterizer& rasterizerProcessor);
+    void update(const MeshRasterizer& rasterizerProcessor) const;
+    virtual void rasterize(const ivec2& imageSize,
+                           std::function<void(Shader&)> setUniforms) const override;
+    virtual bool usesFragmentLists() const override {
+        return !forceOpaque_ && FragmentListRenderer::supportsFragmentLists();
+    }
+    virtual Document getDescription() const override;

```

Comment:
Is is possible to move this outside of the class? Printing info is not really its reponsibility

Reply:
I is, think of it as data. We do the same thing for Image and Volume. since we put it in a port we want to be able to inspect it. We could even add port inspectors for the resterizsers so we could look at the renderings...

Reply:
Renamed now.